src/groupeddataframe/complextransforms.jl

```diff
@@ -397,7 +402,16 @@ function _combine_tables_with_first!(first::Union{AbstractDataFrame,
             return _combine_tables_with_first!(rows, newcols, idx, i, j,
                                                f, gd, incols, colnames, firstmulticol)
         end
-        append!(idx, Iterators.repeated(gdidx[starts[i]], _nrow(rows)))
+        append_const!(idx, gdidx[starts[i]], _nrow(rows))
     end
     return outcols, colnames
 end
+
+@inline function append_const!(idx::Vector{Int}, val, growsize::Int)
```

Comment:
Isn't `val` always `::Integer`?

Also, maybe `@inlined` isn't worth it given the cost of resizing? That could reduce the cost of compiling the caller function, which is specialized on column types.

Reply:
OK - changed. `val` is always `Int` now.